Library/Homebrew/language/python.rb

```diff
@@ -139,11 +139,22 @@ def virtualenv_create(venv_root, python = "python", formula = self)
         venv
       end
 
+      def needs_python?(python)
+        return true if build.with?(python)
+        (requirements.to_a | deps).any? { |r| r.name == python && r.required? }
+      end
+
       # Helper method for the common case of installing a Python application.
       # Creates a virtualenv in `libexec`, installs all `resource`s defined
       # on the formula, and then installs the formula.
-      def virtualenv_install_with_resources
-        venv = virtualenv_create(libexec)
+      def virtualenv_install_with_resources(options = {})
+        python = options[:python]
+        if python.nil?
+          wanted = %w[python python3].select { |py| needs_python?(py) }
+          raise "ambiguous python, please specify" if wanted.size > 1
```

Comment:
This exception message is a bit vague. I wonder if you can improve it and make it more actionable, as it seems not entirely unlikely that it will be presented to contributors.